### PR TITLE
Add documentation for TPT configuration.

### LINKED ISSUE #1979: Docs for relational inheritance mapping
Also mention the Metadata overloads that take a StoreObjectIdentifier

### PR DESCRIPTION
**Pull request by AndriySvyryd**
Fixes #1979

### CODE REVIEW

File: entity-framework/core/modeling/inheritance.md [hunk 1]
```diff
@@ -49,8 +46,41 @@ Finally, the discriminator can also be mapped to a regular .NET property in your
 
 [!code-csharp[Main](../../../samples/core/Modeling/FluentAPI/NonShadowDiscriminator.cs?name=NonShadowDiscriminator&highlight=4)]
 
-## Shared columns
+### Shared columns
```
Comment (smitpatel) [inline entity-framework/core/modeling/inheritance.md:49]: Unrelated to docs question, what happens if user decides to change type of one of them? (let's say the change which is non-destructive)

File: entity-framework/core/modeling/inheritance.md [hunk 2]
```diff
@@ -49,8 +46,41 @@ Finally, the discriminator can also be mapped to a regular .NET property in your
...
 By default, when two sibling entity types in the hierarchy have a property with the same name, they will be mapped to two separate columns. However, if their type is identical they can be mapped to the same database column:
 
 [!code-csharp[Main](../../../samples/core/Modeling/FluentAPI/SharedTPHColumns.cs?name=SharedTPHColumns&highlight=9,13)]
+
+## Table-per-type configuration
+
+> [!NOTE]
+> The table-per-type (TPT) is a new feature in EF Core 5.0. Table-per-concrete-type (TPC) is supported by EF6, but is not yet supported by EF Core.
+
+In the TPT mapping pattern, all types are mapped to individual tables. Properties that belong solely to a base type or derived type are stored in a table that maps to that type. Tables that map to derived types also store a foreign key that joins the derived table with the base table.
```
Comment (smitpatel) [inline entity-framework/core/modeling/inheritance.md:60]: all types in hierarchy.

should it be `all the types`?

File: entity-framework/core/modeling/inheritance.md (hunk 1 above)
Comment (AndriySvyryd) [reply entity-framework/core/modeling/inheritance.md:49]: Validation will fail

File: entity-framework/core/modeling/inheritance.md [hunk 3]
```diff
@@ -49,8 +46,41 @@ Finally, the discriminator can also be mapped to a regular .NET property in your
...
+
+[!code-csharp[Main](../../../samples/core/Modeling/FluentAPI/TPTConfiguration.cs?name=TPTConfiguration)]
+
+EF will create the following following database schema for the the model above.
```
Comment (smitpatel) [inline entity-framework/core/modeling/inheritance.md:64]: Following too much? :trollface:

File: entity-framework/core/modeling/inheritance.md [hunk 4]
```diff
@@ -49,8 +46,41 @@ Finally, the discriminator can also be mapped to a regular .NET property in your
...
+CREATE TABLE [RssBlogs] (
+    [BlogId] int NOT NULL,
+    [RssUrl] nvarchar(max) NULL,
+    CONSTRAINT [PK_RssBlogs] PRIMARY KEY ([BlogId]),
+    CONSTRAINT [FK_RssBlogs_Blogs_BlogId] FOREIGN KEY ([BlogId]) REFERENCES [Blogs] ([BlogId]) ON DELETE NO ACTION
+);
+```
+
+> [!NOTE]
+> If the primary key constraint is renamed the new name will be applied to all tables mapped to the hierarchy, future EF versions will allow renaming the constraint only for a particular table.
```
Comment (smitpatel) [inline entity-framework/core/modeling/inheritance.md:82]: Could link tracking issue here.

File: entity-framework/core/modeling/inheritance.md [hunk 5]
```diff
@@ -49,8 +46,41 @@ Finally, the discriminator can also be mapped to a regular .NET property in your
...
+
+If you are employing bulk configuration you can retrieve the column name for a specific table by calling `GetColumnType`.
```
Comment (smitpatel) [inline entity-framework/core/modeling/inheritance.md:84]: column name from GetColumnType?

Comment (smitpatel) [reply entity-framework/core/modeling/inheritance.md:84]: Does it take parameter? if yes then how to get the value of parameter.
Once we had API docs updated, may be we can just reference actual API.

File: entity-framework/core/modeling/inheritance.md (hunk 1 above)
Comment (smitpatel) [reply entity-framework/core/modeling/inheritance.md:49]: Found my mistake. I thought we merged automatically. I read the docs again and we allow user to configure it so no magical change.

File: entity-framework/core/modeling/inheritance.md (hunk 5 above)
Comment (AndriySvyryd) [reply entity-framework/core/modeling/inheritance.md:84]: The snippet below answers all your questions.
I tried linking, but it seems only types can be linked using xref

Comment (smitpatel) [reply entity-framework/core/modeling/inheritance.md:84]: Snippet does not use `GetColumnType` API.

xref can be used for any API which has page in .NET API explorer. What are you trying to link? GetColumnName overload which takes StoreObjectIdentifier would be missing in API explorer for now.

Comment (AndriySvyryd) [reply entity-framework/core/modeling/inheritance.md:84]: I've tried linking the old GetColumnName - <xref:Microsoft.EntityFrameworkCore.RelationalPropertyExtensions.GetColumnName>

File: entity-framework/core/modeling/inheritance.md [hunk 6]
```diff
@@ -49,8 +46,41 @@ Finally, the discriminator can also be mapped to a regular .NET property in your
...
+    [RssUrl] nvarchar(max) NULL,
+    CONSTRAINT [PK_RssBlogs] PRIMARY KEY ([BlogId]),
+    CONSTRAINT [FK_RssBlogs_Blogs_BlogId] FOREIGN KEY ([BlogId]) REFERENCES [Blogs] ([BlogId]) ON DELETE NO ACTION
+);
+```
+
+> [!NOTE]
+> If the primary key constraint is renamed the new name will be applied to all the tables mapped to the hierarchy, future EF versions will allow renaming the constraint only for a particular table when [issue 19970](https://github.com/dotnet/efcore/issues/19970) is fixed.
+
+If you are employing bulk configuration you can retrieve the column name for a specific table by calling <xref:Microsoft.EntityFrameworkCore.RelationalPropertyExtensions.GetColumnName>.
```
Comment (smitpatel) [inline entity-framework/core/modeling/inheritance.md:84]:
```suggestion
If you are employing bulk configuration you can retrieve the column name for a specific table by calling <xref:Microsoft.EntityFrameworkCore.RelationalPropertyExtensions.GetColumnName%2A>.
```

**Review by smitpatel — APPROVED** [review]

File: entity-framework/core/modeling/inheritance.md (hunk 5 above)
Comment (smitpatel) [reply entity-framework/core/modeling/inheritance.md]: Linked it to "all overload" page which works. In future we should update it to specific API.

**Pull request merged**